### DISCUSSION TITLE
Get info of memory usage of resources

### DISCUSSION THREAD

**Discussion by torsteingrindvik** [gfx-rs/wgpu#4839] [Q&A]
Hi, the context is I was running a Bevy app and also `watch nvidia-smi`, and my GPU was using a few GBs of memory.
Is there a good way to enumerate which textures, buffers, etc. take up space?
In a simple world some tool would be able to list them in order from largest to smallest in order to prioritize where to put effort.

I saw there was something called  [VK_EXT_device_memory_report](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_EXT_device_memory_report.html) which might have helped, but it seems to not be implemented (EDIT: Not in `wgpu` nor `ash`, probably because it isn't ratified?)

Are there other tools which can help out?

**Comment by Wumpf** [comment]
On Windows and Linux [RenderDoc](https://renderdoc.org/) is usually a great tool to debug this sort of thing

**Reply by torsteingrindvik** [reply]
I've used RenderDoc for frame debugging but I don't think it can aggregate mem usage?

I open a capture of my app and looked around, and I can see sizes of individual textures and buffers in use for a given frame but I'm looking for something like a CPU heap profiler but for the GPU.

**Reply by Wumpf** [reply]
hum, you're right. Thought RenderDoc can do that, but it doesn't :(

Looke like VK_EXT_device_memory_report is only implemented by a [bunch of mobile chips](https://www.vulkan.gpuinfo.org/listdevicescoverage.php?extension=VK_EXT_device_memory_report&platform=all) today.

I think the only reliable thing you can try is to use a vendor specific tool: Actual memory used by textures etc. is actually quite tricky since gpus often do loss-less compression on one hand and device specific padding on the other, so only a vendor specific tool like NSight will give reliable numbers.

Otherwise I reckon you might still try to use RenderDoc to capture all buffers & textures and then use that information to do an estimate of the amount of memory used by processing the data (e.g. by using the [python api](https://renderdoc.org/docs/python_api/renderdoc/resources.html) to get all the necessary data out 🤔 )
